### PR TITLE
Fill in largest tax rate applied for Anne Arundel county in MD-502 line 28 pdf and XML

### PR DESCRIPTION
**Pull request by embarnard**
## Link to pivotal/JIRA issue
- e.g. [link](https://codeforamerica.atlassian.net/browse/FYST-1453)
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- Return largest local tax rate for Anne Arundel county in MD 502 
- previously because this county uses a progressive tax rate we didn't fill in line 28 on MD 502 but now we are
## How to test?
- Create a Maryland intake in the Anne Arundel county and make sure that line 28 is filled out and `LocalTaxRate` in the XML
## Screenshots (for visual changes)
- After
Screenshot 2024-12-16 at 10 14 10 PM
Screenshot 2024-12-16 at 10 14 49 PM

### CODE REVIEW

File: spec/lib/efile/md/md502_calculator_spec.rb [hunk 1]
```diff
@@ -1263,15 +1306,36 @@
...
+        it "calculates the local tax progressively" do
+          expect(instance.lines[:MD502_LINE_28_LOCAL_TAX_AMOUNT].value).to eq(8375)
+        end
+      end
+
+      context "when income is 450,000" do
+        let(:income) { 450_000 }
+        it "calculates the local tax progressively" do
+          # (0.027 * 50,000) + (0.0281 * 350,000) + (.032 * 50,000) = 12785
+          expect(instance.lines[:MD502_LINE_28_LOCAL_TAX_AMOUNT].value).to eq(127_85)
```
Comment (arinchoi03) [inline spec/lib/efile/md/md502_calculator_spec.rb:1329]: [Dust] i'm not really certain why we use underscores for integers 😓 I'm guessing for readability? but should this be `12_785`?

File: spec/lib/efile/md/md502_calculator_spec.rb [hunk 2]
```diff
@@ -1300,9 +1364,9 @@
 
     context "when county is Anne Arundel" do
       let(:county) { "Anne Arundel" }
-      it "uses 0.0270 as the local tax rate in the formula" do
-        # (0.027 * 10) * 1001
-        expect(instance.lines[:MD502_LINE_29].value).to eq(270)
+      it "uses 0.0281 as the local tax rate in the formula" do
+        # (0.0281 * 10) * 1001
+        expect(instance.lines[:MD502_LINE_29].value).to eq(281)
```
Comment (arinchoi03) [inline spec/lib/efile/md/md502_calculator_spec.rb:1369]: should we flag to acceptance testers that the `EarnedIncomeCredit` value utilizes the highest local rate for Anne Arundel county now? [For the original story it seemed like we were saying to use 0.270 ](https://codeforamerica.atlassian.net/browse/FYST-770)

[Added a jira question here](https://codeforamerica.atlassian.net/browse/FYST-1453?focusedCommentId=14143)

**Review by arinchoi03 — APPROVED** [review]
neat! 

I just had one q for Product so I added that to the ticket.

File: spec/lib/efile/md/md502_calculator_spec.rb (hunk 1 above)
Comment (embarnard) [reply spec/lib/efile/md/md502_calculator_spec.rb:1329]: @arinchoi03 yes it's just readability! Its equal to a comma, so you're totally right it should be `12_785`

File: spec/lib/efile/md/md502_calculator_spec.rb (hunk 2 above)
Comment (embarnard) [reply spec/lib/efile/md/md502_calculator_spec.rb:1369]: @arinchoi03 yes thank you for flagging!

**Pull request merged**